File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static int n, m, count;
    static int[][] board;
    static boolean[] visited;
    
    static void input() throws IOException {
        n = Integer.parseInt(br.readLine());
        m = Integer.parseInt(br.readLine());
        board = new int[n + 1][n + 1];
        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());
            board[a][b] = 1;
            board[b][a] = 1;
        }
    }
    
    static void recur(int i) {
        visited[i] = true;
        for (int j = 1; j < n + 1; j++) {
            if (board[i][j] != 1 || visited[j]) {
                continue;
            }
            count++;
            recur(j);
        }
    }
    
    static void process() throws IOException {
        count = 0;
        visited = new boolean[n + 1];
        recur(1);
        bw.write(count + "\n");
    }
    
    static void output() throws IOException {
        bw.flush();
        bw.close();
    }
    
    public static void main(String[] args) throws IOException {
        input();
        process();
        output();
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

input = sys.stdin.readline


def recur(i: int):
    global count, visited

    visited[i] = True
    for j in range(n + 1):
        if board[i][j] != 1 or visited[j]:
            continue
        count += 1
        recur(j)


n = int(input())
m = int(input())
board = [[0 for _ in range(n + 1)] for _ in range(n + 1)]
for _ in range(m):
    a, b = map(int, input().split())
    board[a][b] = 1
    board[b][a] = 1

count = 0
visited = [False for _ in range(n + 1)]
recur(1)

print(count)